ui/raidboss/emulator/overrides/RaidEmulatorOverlayApiHook.ts

```diff
@@ -37,53 +38,62 @@ export default class RaidEmulatorOverlayApiHook {
+  msg: OverlayHandlerRequests['getCombatants'],
+  curEnc: AnalyzedEncounter,
+  tracker: CombatantTracker,
+  timestamp: number): void {
+    const combatants: PluginCombatantState[] = [];
+    const ids = msg.ids ?? [];
+    const names = msg.names ?? [];
+    const hasIds = ids.length > 0;
+    const hasNames = names.length > 0;
 
-      for (const [id, combatant] of Object.entries(tracker.combatants)) {
-        // If this combatant didn't exist at this point, skip them
-        const firstStateStamp = combatant.significantStates[0];
-        const lastStateStamp = combatant.significantStates.slice(-1)[0];
-        if (!firstStateStamp || !lastStateStamp)
-          continue;
-        if (firstStateStamp > timestamp || lastStateStamp < timestamp)
-          continue;
+    for (const [id, combatant] of Object.entries(tracker.combatants)) {
+      // If this combatant didn't exist at this point, skip them
+      const firstStateStamp = combatant.significantStates[0];
+      const lastStateStamp = combatant.significantStates.slice(-1)[0];
+      if (!firstStateStamp || !lastStateStamp)
+        continue;
+      if (firstStateStamp > timestamp || lastStateStamp < timestamp)
+        continue;
 
-        const idNum = parseInt(id, 16);
-        // nextSignificantState is a bit inefficient but given that this isn't run every tick
-        // we can afford to be a bit inefficient for readability's sake
-        const combatantState = {
-          ID: idNum,
-          Name: combatant.name,
-          Level: combatant.level,
-          Job: combatant.jobId,
-          ...combatant.nextSignificantState(timestamp).toPluginState(),
-        };
-        if (!hasIds && !hasNames)
-          combatants.push(combatantState);
-        else if (hasIds && ids.includes(idNum))
-          combatants.push(combatantState);
-        else if (hasNames && names.includes(combatant.name))
-          combatants.push(combatantState);
+      const idNum = parseInt(id, 16);
+      // nextSignificantState is a bit inefficient but given that this isn't run every tick
+      // we can afford to be a bit inefficient for readability's sake
+      const combatantState = {
+        ID: idNum,
+        Name: combatant.name,
+        Level: combatant.level,
+        Job: combatant.jobId,
+        ...combatant.nextSignificantState(timestamp).toPluginState(),
+      };
+      if (!hasIds && !hasNames)
+        combatants.push(combatantState);
+      else if (hasIds && ids.includes(idNum))
+        combatants.push(combatantState);
+      else if (hasNames && names.includes(combatant.name))
+        combatants.push(combatantState);
+    }
+    // @TODO: Move this to track properly on the Combatant object
+    combatants.forEach((c) => {
+      const lines = curEnc.encounter.logLines
+        .filter((l) => l.decEvent === 3 && isLineEventSource(l) && parseInt(l.id, 16) === c.ID);
+      const baseLine = lines[0];
+      if (baseLine) {
+        const line = baseLine as LineEvent0x03;
```

Comment:
Not for this patch, but maybe line events should have `isLineEvent0x03` type guard functions?

Reply:
I was thinking of revisiting LineEvent and making `decEvent` and/or `hexEvent` a union type, and replacing all instances of `LineEvent` with a `LineEventAny` union of all `LineEvent` types, so that just checking the `decEvent`/`hexEvent` properties is sufficient for ts to narrow the type to the right subclass.